mk/python.mk

```diff
@@ -1,12 +1,35 @@
-PYTHON = python3.5
-PYTHON := $(shell which $(PYTHON))
+# check Python version
+PYTHON := $(shell which "python")
+PYTHON_COMPATIBLE = python3.5
+
 ifndef PYTHON
-$(error "python3.5 is required.")
+    $(error "Python is not executable, try to check the PATH environment variable or install $(PYTHON_COMPATIBLE) package.")
+endif # PYTHON
+
+ifdef PYTHON
+    PY_CHECK_VERSION := $(shell python -c "print(__import__('sys').version_info[0:2])")
+
+ifneq ($(PY_CHECK_VERSION), (3, 5))
+    PYTHON_COMPATIBLE_PATH := $(shell which $(PYTHON_COMPATIBLE))
+
+    ifdef PYTHON_COMPATIBLE_PATH
+        $(error "Found $(PYTHON_COMPATIBLE) in current environment, but it should be setting as \
+            default interpreter, try to create a symbolic link for $(PYTHON_COMPATIBLE_PATH)")
```

Comment:
Don't suggest to make symbolic link. Instead, configure via environment variables.